blockdev/src/blockdev.rs

```diff
@@ -217,13 +237,59 @@ impl LoopbackDevice {
         self.dev.as_deref().unwrap()
     }
 
+    /// Spawn a cleanup helper process that will clean up the loopback device
+    /// if the parent process dies unexpectedly
+    fn spawn_cleanup_helper(device_path: &str) -> Result<LoopbackCleanupHandle> {
+        use std::process::{Command, Stdio};
+
+        // Get the path to our own executable
+        let self_exe =
+            std::fs::read_link("/proc/self/exe").context("Failed to read /proc/self/exe")?;
+
+        // Create the helper process
+        let mut cmd = Command::new(self_exe);
+        cmd.args([
+            "loopback-cleanup-helper",
+            "--device",
+            device_path,
+            "--parent-pid",
+            &std::process::id().to_string(),
+        ]);
+
+        // Set environment variable to indicate this is a cleanup helper
+        cmd.env("BOOTC_LOOPBACK_CLEANUP_HELPER", "1");
+
+        // Set up stdio to redirect to /dev/null
+        cmd.stdin(Stdio::null());
+        cmd.stdout(Stdio::null());
+        cmd.stderr(Stdio::null());
+
+        // Spawn the process
+        let child = cmd
+            .spawn()
+            .context("Failed to spawn loopback cleanup helper")?;
+
+        Ok(LoopbackCleanupHandle {
+            helper_pid: child.id(),
+        })
+    }
+
     // Shared backend for our `close` and `drop` implementations.
     fn impl_close(&mut self) -> Result<()> {
         // SAFETY: This is the only place we take the option
         let Some(dev) = self.dev.take() else {
             tracing::trace!("loopback device already deallocated");
             return Ok(());
         };
+
+        // Kill the cleanup helper since we're cleaning up normally
+        if let Some(cleanup_handle) = self.cleanup_handle.take() {
+            // Kill the helper process since we're doing normal cleanup
+            let _ = std::process::Command::new("kill")
```

Comment:
See above, we can directly send a signal instead of invoking an external command.

Also, crucially by holding a reference to the `Child` we avoid pid reuse problems if the child exits early.